### PR TITLE
Develop verify for mrm

### PR DESCRIPTION
**Pull request by ajit3190**

### CODE REVIEW

**Review by pnabutovsky — CHANGES_REQUESTED** [review]
When creating new application strings under config/locales only modify the English files. Please do not edit the other language files: they are maintained with Transifex. Primero automatically provides the English string (or sometimes a language fallback chain) if a translation is missing.

File: app/javascript/components/record-form/form/subforms/styles.css [hunk 1]
```diff
@@ -1,3 +1,14 @@
+.verifiedSpan{
+  color: white;
```
Comment (jtoliver-quoin) [inline app/javascript/components/record-form/form/subforms/styles.css:2]: Use `var(--c-white)` instead

File: app/javascript/components/record-form/form/subforms/styles.css [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+  height: 25px;
+  font-size: 13px;
```
Comment (jtoliver-quoin) [inline app/javascript/components/record-form/form/subforms/styles.css:4]: use `var(--fs-13)`

File: app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/component.jsx [hunk 3]
```diff
@@ -1,18 +1,70 @@
...
 
 import ViolationTitle from "../violation-title";
 import css from "../../../styles.css";
 
 import { NAME } from "./constants";
 import { getViolationTallyLabel } from "./utils";
 
-const Component = ({ fields, values, locale, displayName, index, collapsedFieldValues }) => {
-  const currentValues = values[index];
+import { Button } from "@material-ui/core";
```
Comment (jtoliver-quoin) [inline app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/component.jsx:13]: Run`npm run lint` and `npm run test` and fix any lint issues or test

File: app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/component.jsx [hunk 4]
```diff
@@ -1,18 +1,70 @@
...
+import { useParams} from 'react-router-dom';
+import { usePermissions } from "../../../../../../permissions";
+import { RECORD_ACTION_ABILITIES } from "../../../../../../record-actions/constants";
+import VerifySelect from "./select";
+import { useI18n } from "../../../../../../i18n";
+import { toServerDateFormat } from "../../../../../../../libs";
 
+const Component = ({ fields, values, locale, displayName, index, collapsedFieldValues, mode }) => {
+  const currentValues = values[index];
+  const [selectedIndex, setSelectedIndex] = useState(null);
```
Comment (jtoliver-quoin) [inline app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/component.jsx:24]: group useState variables together

File: app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/component.jsx [hunk 5]
```diff
@@ -1,18 +1,70 @@
...
+  const dispatch = useDispatch();
+  const [verifyModal, setVerifyModal] = useState(false);
+  const [verificationValue, setVeficationValue] = useState(currentValues?.ctfmr_verified || '') //Dropdown selected state
   const violationTally = getViolationTallyLabel(fields, currentValues, locale);
+  const verifyParams = useParams();
+  const handleOpenVerifyModal = (index,event) => {    //  To open verify dialog confirmation popup
```
Comment (jtoliver-quoin) [inline app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/component.jsx:30]: Add a newline above

File: app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/component.jsx [hunk 6]
```diff
@@ -1,18 +1,70 @@
...
+  const {
+    canVerify
+  } = usePermissions("incidents", RECORD_ACTION_ABILITIES);   //  To check permission to do verify violations
+
+  useEffect(() => { // Changing dropdown select value when backend data updated
+    setVeficationValue(currentValues?.ctfmr_verified)
+  }, [currentValues?.ctfmr_verified]);
+
+  const violationType = currentValues.type    //  To get the violation type through index
+  const handleOk = () => {    //  To update the verify status to Verified
```
Comment (jtoliver-quoin) [inline app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/component.jsx:49]: Add a new line above

File: app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/component.jsx [hunk 7]
```diff
@@ -1,18 +1,70 @@
...
+    event.stopPropagation();
+    setSelectedIndex(index);
+    setVerifyModal(true);
+  };
+
+  const cancelVerifyHandler = () => {   //  To close veify dialog popup
+    setVerifyModal(false);
+    setSelectedIndex(null);
+  };
+  const {
```
Comment (jtoliver-quoin) [inline app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/component.jsx:40]: Add a newline above

File: app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/styles.css [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  border-radius: 4px;
+  border-bottom: none;
+}
+
+.verifySelectComponentSelect, .verifySelectComponentSelect:focus {
+  border-radius: 4px;
+}
+
+.verifySelectComponent[aria-expanded="true"], .verifySelectComponent:focus {
+  outline: 4px solid rgba(31, 176, 214, 0.4);
```
Comment (jtoliver-quoin) [inline app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/styles.css:17]: The color used here should be added to our root css vars or use one of the existing css vars

Comment (jtoliver-quoin) [reply app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/styles.css:17]: This color is a little odd. Could you attach a screenshot of this?

**Review by jtoliver-quoin — CHANGES_REQUESTED** [review]

**Pull request merged**